File: Project/backgammon-app/frontend/src/types/detection.ts
export interface BoundingBox {
  x: number
  y: number
  width: number
  height: number
}

export interface CheckerCount {
  white: number
  black: number
}

export interface FrameResult {
  turn: number
  player: string
  dice: number[]
  cube: number
  points: Record<string, CheckerCount>
  bar: CheckerCount
  borne_off: CheckerCount
}

export interface DetectionResponse {
  video_filename: string
  total_frames_analyzed: number
  fps: number
  duration_sec: number
  frames: FrameResult[]
  error?: string
}

File: Project/backgammon-app/backend/app/schemas/detection.py
from pydantic import BaseModel
from typing import Dict, List, Optional


class CheckerCount(BaseModel):
    white: int
    black: int


class FrameResult(BaseModel):
    turn: int
    player: str
    dice: List[int]
    cube: int
    points: Dict[str, CheckerCount]
    bar: CheckerCount
    borne_off: CheckerCount


class DetectionResponse(BaseModel):
    video_filename: str
    total_frames_analyzed: int
    fps: float
    duration_sec: float
    frames: List[FrameResult]
    error: Optional[str] = None
